modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/util/ExtensionsTestUtils.java

```diff
@@ -166,4 +171,36 @@ public static File createManifestFileIfNecessary(File targetDirectory, Manifest
         return manifestFile;
     }
 
+    /**
+     * Receives to {@link String} representation of two XML
+     * files and verify that they are semantically equivalent
+     *
+     * @param expected the reference content
+     * @param actual   the actual content
+     * @throws Exception if comparison fails
+     */
+    public static void compareXML(String expected, String actual) throws Exception
+    {
+        XMLUnit.setNormalizeWhitespace(true);
+        XMLUnit.setIgnoreWhitespace(true);
+        XMLUnit.setIgnoreComments(true);
+        XMLUnit.setIgnoreAttributeOrder(true);
+
+        Diff diff = XMLUnit.compareXML(expected, actual);
+        if (!(diff.similar() && diff.identical()))
+        {
+            System.out.println(actual);
+            DetailedDiff detDiff = new DetailedDiff(diff);
+            @SuppressWarnings("rawtypes")
+            List differences = detDiff.getAllDifferences();
+            StringBuilder diffLines = new StringBuilder();
+            for (Object object : differences)
+            {
+                Difference difference = (Difference) object;
+                diffLines.append(difference.toString() + '\n');
+            }
+
+            throw new IllegalArgumentException(String.format("The Output for extension [%s] schema was not the expected:", diffLines.toString()));
```

Comment:
actually, this method is generic and shouldn't reference any extension at all. Changed